addons/mail/static/src/core_ui/message.js

```diff
@@ -315,15 +316,24 @@ export class Message extends Component {
     }
 
     onClickPin() {
-        const prompt = this.message.pinned_at
-            ? _t("Are you sure you want to remove this pinned message?")
-            : _t(
-                  "The following message will be pinned to the channel. Are you sure you want to continue?"
-              );
+        const thread = this.message.originThread;
         this.env.services.dialog.add(MessageConfirmDialog, {
+            confirmColor: this.message.pinned_at ? "btn-danger" : undefined,
+            confirmText: this.message.pinned_at
+                ? _t("Yes, remove it please")
+                : _t("Oh yeah. Pin it"),
             message: this.message,
             messageComponent: Message,
-            prompt,
+            prompt: this.message.pinned_at
+                ? _t("Enough of that, I guess. You sure you want to remove this pinned message?")
+                : sprintf(
+                      _t(
+                          "Hey, just double checking that you want to pin this message to %(conversation_name)s for posterity and greatness?"
```

Comment:
I know this is copy pasta text, but maybe if we're allowed to write something else then we could use the slightly less cringe:

```suggestion
                ? _t("Well nothing lasts forever, but are you sure you want to unpin this message?")
                : sprintf(
                      _t(
                          "You sure want this message pinned to %(conversation_name)s forever and ever?"
```

unless we're afraid of upsetting people going through sad periods in life (i.e. death, relationship ending, etc.) :thinking: 

Also both "remove" and "unpin" work in this case, but either way the button should match